application/src/main/java/org/opentripplanner/transit/api/request/TripTimeOnDateRequestBuilder.java

```diff
@@ -0,0 +1,132 @@
+package org.opentripplanner.transit.api.request;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Collection;
+import java.util.Comparator;
+import java.util.List;
+import javax.annotation.Nullable;
+import org.opentripplanner.model.TripTimeOnDate;
+import org.opentripplanner.transit.api.model.FilterValues;
+import org.opentripplanner.transit.model.basic.TransitMode;
+import org.opentripplanner.transit.model.framework.FeedScopedId;
+import org.opentripplanner.transit.model.site.StopLocation;
```

Comment:
A request class should depend on a `StopLocation`, that is strange.

Reply:
We talked about it in the dev meeting and the reason is that many methods in `TransitService` already use the `StopLocation` as the input type. Many underlying indices also use it as the key.